src/finch/symbolic/term.py

```diff
@@ -28,15 +28,9 @@ def head(cls) -> Callable[..., Self]:
         raise NotImplementedError
 
     @abstractmethod
-    def children(self) -> list[Term]:
+    def children(self) -> list[Any]:
```

Comment:
This was done as some terms, e.g. `Immediate`, `Alias`, ... return `str` or `Any`.